playground/infrastructure/helper.py

```diff
@@ -44,29 +53,187 @@ def find_examples(work_dir: str) -> List[Example]:
         categories:
             - category-1
             - category-2
+    If some example contain beam tag with incorrect format raise an error.
 
     Args:
         work_dir: directory where to search examples.
+        categories_path: path to the file with all supported categories.
 
     Returns:
         List of Examples.
     """
-    # TODO [BEAM-13135] Implement
-    pass
+    failed = False
+    examples = []
+    for root, _, files in os.walk(work_dir):
+        for filename in files:
+            filepath = os.path.join(root, filename)
+            extension = filepath.split(os.extsep)[-1]
+            if extension in Config.SUPPORTED_SDK:
+                tag = get_tag(filepath)
+                if tag:
+                    if _validate(tag, categories_path) is False:
+                        logging.error(filepath + "contains beam playground tag with incorrect format")
+                        failed = True
+                    else:
+                        examples.append(_get_example(filepath, filename))
+    if failed:
+        raise ValueError("some of the beam examples contain beam playground tag with incorrect format")
+    return examples
 
 
 def get_statuses(examples: List[Example]):
     """
-    Receive statuses for examples and update example.status and example.pipelineId
+    Receive statuses for examples and update example.status and example.pipeline_id
 
     Use client to send requests to the backend:
     1. Start code processing.
     2. Ping the backend while status is STATUS_VALIDATING/STATUS_PREPARING/STATUS_COMPILING/STATUS_EXECUTING
-    Update example.pipelineId with resulting pipelineId.
+    Update example.pipeline_id with resulting pipelineId.
     Update example.status with resulting status.
 
     Args:
         examples: beam examples for processing and updating statuses.
     """
     # TODO [BEAM-13267] Implement
     pass
+
+
+def get_tag(filepath):
+    """
+    Parse file by filepath and find beam tag
+
+    Args:
+        filepath: path of the file
+
+    Returns:
+        If file contains tag, returns tag as a map.
+        If file doesn't contain tag, returns None
+    """
+    add_to_yaml = False
+    yaml_string = ""
+
+    with open(filepath) as parsed_file:
+        lines = parsed_file.readlines()
+
+    for line in lines:
+        line = line.replace("// ", "").replace("# ", "")
+        if add_to_yaml is False:
+            if line == BEAM_PLAYGROUND_TITLE:
+                add_to_yaml = True
+                yaml_string += line
+        else:
+            yaml_with_new_string = yaml_string + line
+            try:
+                yaml.load(yaml_with_new_string, Loader=yaml.SafeLoader)
+                yaml_string += line
+            except Exception:
```

Comment:
Let's except only the exceptions that `yaml.load` throws

Reply:
Changed.